pkg/plugins/gateway/algorithms/util.go

```diff
@@ -60,3 +61,29 @@ func SelectRandomPodAsFallback(ctx *types.RoutingContext, pods []*v1.Pod, random
 	}
 	return targetPod, nil
 }
+
+// newPod creates a new Pod object for testing purposes.
+// It constructs a Pod with basic configuration including name, IP address, ready status, and labels.
+// returns a pointer to the created Pod object
+func newPod(name, ip string, ready bool, labels map[string]string) *v1.Pod {
```

Comment:
could you move this method to test file since it's test specific?

Reply:
if other tests use this method, it's their responsibility to refactor to common test utils. :D